### PR TITLE
Update Japanese translation

### PR DESCRIPTION
**Pull request by gary-ts**
Recommended Japanese font size for 1080p monitors:
Lines: 36
Infos text size: 0.80

### CODE REVIEW

File: language/Japanese/strings.po [hunk 1]
```diff
@@ -5584,82 +5610,82 @@ msgstr "連射"
...
 msgctxt "plugin-autofire"
 msgid "Input"
-msgstr ""
+msgstr "通常ボタン"
 
 #: plugins/autofire/autofire_menu.lua:159
 msgctxt "plugin-autofire"
 msgid "Hotkey"
-msgstr ""
+msgstr "連射ボタン"
```
Comment (cuavas) [inline language/Japanese/strings.po]: “ボタン” is overly specific.

File: language/Japanese/strings.po [hunk 2]
```diff
@@ -5584,82 +5610,82 @@ msgstr "連射"
...
 msgid "Autofire buttons"
-msgstr ""
+msgstr "連射ボタン"
 
 #: plugins/autofire/autofire_menu.lua:75
 #, lua-format
 msgctxt "plugin-autofire"
 msgid "Press %s to delete"
-msgstr ""
+msgstr "%s を押すと削除"
```
Comment (cuavas) [inline language/Japanese/strings.po]: The grammar doesn’t seem right for this.

File: language/Japanese/strings.po [hunk 3]
```diff
@@ -5518,24 +5544,24 @@ msgstr "一致ブロック"
...
 #: plugins/cheatfind/init.lua:907
 #, lua-format
 msgid "Test Cheat %08X_%02X"
-msgstr ""
+msgstr "チート %08X_%02X をテスト"
 
 #: plugins/cheatfind/init.lua:947
 msgid "Cheat engine not available"
-msgstr "チートエンジンが利用できません"
+msgstr "チートエンジンを利用できません"
```
Comment (cuavas) [inline language/Japanese/strings.po]: This change is incorrect.

File: language/Japanese/strings.po [hunk 4]
```diff
@@ -5518,24 +5544,24 @@ msgstr "一致ブロック"
...
 #: plugins/cheatfind/init.lua:798
 msgid "All"
-msgstr "全て"
+msgstr "すべて"
 
 #: plugins/cheatfind/init.lua:907
 #, lua-format
 msgid "Test Cheat %08X_%02X"
-msgstr ""
+msgstr "チート %08X_%02X をテスト"
```
Comment (cuavas) [inline language/Japanese/strings.po]: This is not an imperative.

File: language/Japanese/strings.po [hunk 5]
```diff
@@ -5312,197 +5338,197 @@ msgstr "タイプ"
...
+msgstr "チートを cheat.simple に追加しました"
 
 #: plugins/cheatfind/init.lua:483
 msgid ""
 "Unable to write file\n"
 "Ensure that cheatpath folder exists"
 msgstr ""
 "ファイルに書き込めません\n"
-"cheatpathフォルダが存在しているか確認してください"
+"cheat パスのフォルダが存在するか確認してください"
```
Comment (cuavas) [inline language/Japanese/strings.po]: This change is incorrect – `cheatpath` is the name of an option as it appears on the command line or in INI files.

**Review by cuavas — COMMENTED** [review]
I’m sorry, but this update doesn’t meet quality standards.  It makes terminology inconsistent in Japanese.  We already have consistency issues with English terminology, and we really don’t need worse problems in translations.  It carries across problematic terminology that needs to be fixed in English (e.g. “slot” in the cheat finder).  It makes arbitrary changes that aren’t improvements.  I’ve commented on a few things that are problematic, but going through it, there’s just too much.

**Comment by gary-ts** [comment]
Thank you for checking my translation.

> It makes terminology inconsistent in Japanese.

Maybe you can translate English into French (or Latin languages) consistently.
An English word often has some meanings for Japanese. We can't translate it to one consistent word.

I personally think "inconsistent" translation is not bad.
Descriptive translation is sometimes inconsistent and "inaccurate", but it's easy to understand.

"coding style" and "natural languages" are different.
You might want consistent coding style as a maintainer, but users need translation that is easy to understand.

You lost one translator.
Please be tolerant to translators and their translation. They are few (especially in some languages) and not experts on emulation development.

> "PO-Revision-Date: 2016-02-23 01:39+0900\n"

2022-01-08: Rejected.

I needed to make a huge Japanese translation update.
You might like DeepL translation better. It will make "consistent" translation.

**Comment by Firehawke** [comment]
> I’m sorry, but this update doesn’t meet quality standards. It makes terminology inconsistent in Japanese. We already have consistency issues with English terminology, and we really don’t need worse problems in translations. It carries across problematic terminology that needs to be fixed in English (e.g. “slot” in the cheat finder). It makes arbitrary changes that aren’t improvements. I’ve commented on a few things that are problematic, but going through it, there’s just too much.

This is something we need to fix in English first, I think. It may be time to come up with a new terminology structure to try to reduce confusion.

**Comment by cuavas** [comment]
This change introduces inconsistency in Japanese in places where there’s none in English, though.

For better or worse, MAME calls the thing you map an emulated input to a “sequence” in English consistently, but this change called it “command”, “key” and “button” in different messages in Japanese, even where they would show up in the same context (e.g. when configuring an input macro activation sequence).  We need to actually decide on what it should be called in Japanese and use that consistently throughout the interface.

It’s overly literal in places, e.g. the poorly named “slot” in the cheat finder is translated literally, which is likely to cause confusion, given MAME usually uses “slot” for the slot/card idiom.  This needs to be fixed in English when we decide on standard terminology for it (the cheat finder needs a rewrite, and not just for the poorly worded UI), but that doesn’t stop translations from using better terminology (e.g. it’s translated more like “memory snapshot” Czech and Slovak).

It’s overly specific in places, for example calling inputs “buttons” specifically in the input macro plugin.  The input macro plugin is not only used with buttons.  Even the examples in the documentation demonstrate use of the plugin with joystick inputs.  Translating “inputs” as “buttons” is overly specific and misleading.

Various things are translated without regard for context, e.g. translating “Test cheat” as an imperative where “Test” is used as an adjective or specifier, or attempting to translate “cheatpath” when it’s intentionally preserved literally because it’s the name of an option on the command line or in INI files.

Various Japanese terminology that was already agreed on is changed seemingly arbitrarily.

There are simply too many issues with this to be worth raising them individually.

**Comment by gary-ts** [comment]
I wrote [README](https://github.com/gary-ts/mame#cautions) for unofficial Japanese translation of MAME.

## Cautions

Some explanations have been added and words have been changed to make it easier to understand for people who are new to MAME. This translation is not a direct translation and is not rigorous. The developers of MAME seem to want a literal translation, so the Pull Request has been [rejected](https://github.com/mamedev/mame/pull/9104). After sending one reply, I have not seen the thread. It's sad to see a big update rejected so soon. :-)

The language can be changed at any time, so once you understand the content through the Japanese translation, you can change it back to English for rigor.

## About the original MAME messages and translations

The current menu items in MAME are too simple to understand their functions. To use an analogy, it is like a command line option, and translating only that is unnatural and difficult to understand the function. To understand the function of command line options, you need to read the help.

If a menu item just says "Cheat", you don't know if it means "Enable cheat", "Toggle cheat", or "Cheat settings". In my case, I couldn't tell the difference between "Synchronized Refresh" and "Wait Vertical Sync" or the meaning of "Coin impulse" just by looking at the item. The translator is not an emulator developer.

In MAME, when you select a menu item, an explanation is displayed below it. I have added some of its contents to the translation. Note that the explanation displayed below is not translatable (e.g. "use bilinear filtering when scaling emulated video").

If there is no description in the menu item, I have added part of the description from [MAME Documentation](https://docs.mamedev.org/) to the translation. The MAME UI has a strict character limit, so it was difficult to summarize.

Some overly long messages have been shortened. For example, the following message took up the entire width of the screen, so I shortened it a bit.

This driver requires images to be loaded in the following device(s): "cartridge" ->

Load images in the following device(s): "cartridge

It's not good that the translation differs from the original message, but I originally started updating the translation because I didn't know how it would work just by reading the original message. MAME's current original message is difficult to understand the function if translated as is, so I prioritized clarity.

**Pull request closed without merging**